File: frontend/src/app/Types/Dropdown.ts
import { IController } from './IController';

export class Dropdown implements IController {
  constructor(
    public items: Array<any>,
    public label: string = '',
    public title: string = '',
    public items2: Array<any> | null = null,
    public label2: string = ''
  ) {}

  isDropdown() {
    return true;
  }

  isSlider() {
    return false;
  }

  isCheckbox() {
    return false;
  }
}

File: RestApi/Types/Dropdown.py
class Dropdown:
    def __init__(self, title="", label="", items=[], label2="", items2=None, type=""):
        self.title = title
        self.items = items
        self.label = label
        self.items2 = items2
        self.label2 = label2
        self.dropdown_type = type
        self.type = "dropdown"

    def to_dict(self):
        return dict(self.__dict__)
